apps/dashboard/src/components/activity/activity-filters.tsx

```diff
@@ -0,0 +1,182 @@
+import { useEffect, useMemo } from 'react';
+import { ChannelTypeEnum } from '@novu/shared';
+import { useFetchWorkflows } from '../../hooks/use-fetch-workflows';
+import { useForm } from 'react-hook-form';
+import { Form, FormItem, FormField } from '../primitives/form/form';
+import { Button } from '../primitives/button';
+import { FacetedFormFilter } from '../primitives/form/faceted-filter/facated-form-filter';
+import { CalendarIcon } from 'lucide-react';
+
+export interface IActivityFilters {
+  onFiltersChange: (filters: IActivityFiltersData) => void;
+  initialValues: IActivityFiltersData;
+  onReset?: () => void;
+}
+
+export interface IActivityFiltersData {
+  dateRange: string;
+  channels: ChannelTypeEnum[];
+  workflows: string[];
+  transactionId: string;
+  subscriberId: string;
+}
+
+const DATE_RANGE_OPTIONS = [
+  { value: '24h', label: 'Last 24 hours' },
+  { value: '7d', label: 'Last 7 days' },
+  { value: '30d', label: 'Last 30 days' },
+];
+
+const CHANNEL_OPTIONS = [
+  { value: ChannelTypeEnum.SMS, label: 'SMS' },
+  { value: ChannelTypeEnum.EMAIL, label: 'Email' },
+  { value: ChannelTypeEnum.IN_APP, label: 'In-App' },
+  { value: ChannelTypeEnum.PUSH, label: 'Push' },
+];
+
+export const defaultActivityFilters: IActivityFiltersData = {
+  dateRange: '30d',
+  channels: [],
+  workflows: [],
+  transactionId: '',
+  subscriberId: '',
```

Comment:
What about after and before dates?

Reply:
They are specified as dateRange in the UI, before and after is a backend implementation